### PR TITLE
Clean up sln reader/writer

### PR DESCRIPTION
**Pull request by jgoshi**
@piotrpMSFT @livarcocc @krwq @blackdwarf 

Round one of cleaning up the source we snapped. I also plan to eventually implement much more comprehensive tests for this. See issue #5004

### CODE REVIEW

File: src/Microsoft.DotNet.ProjectJsonMigration/ProjectDependencyFinder.cs [hunk 1]
```diff
@@ -372,7 +372,7 @@ internal static List<string> GetSolutionPaths(SlnFile solutionFile)
             return (solutionFile == null)
                 ? new List<string>()
                 : new List<string>(solutionFile.Projects.Select(p =>
-                      Path.Combine(solutionFile.BaseDirectory.FullPath, Path.GetDirectoryName(p.FilePath))));
+                      Path.Combine(solutionFile.BaseDirectory, Path.GetDirectoryName(p.FilePath))));
```
Comment (krwq) [inline src/Microsoft.DotNet.ProjectJsonMigration/ProjectDependencyFinder.cs]: what is this doing in this PR?

File: src/dotnet/commands/dotnet-migrate/MigrateCommand.cs [hunk 2]
```diff
@@ -115,7 +115,7 @@ private void UpdateSolutionFile(MigrationReport migrationReport)
             foreach (var project in _slnFile.Projects)
             {
                 var projectDirectory = Path.Combine(
-                    _slnFile.BaseDirectory.FullPath,
```
Comment (krwq) [inline src/dotnet/commands/dotnet-migrate/MigrateCommand.cs]: same question

**Comment by jgoshi** [comment]
test Windows_NT x64 Release Build

**Comment by jgoshi** [comment]
test Fedora23 x64 Debug Build

**Comment by jgoshi** [comment]
test Windows_NT x86 Debug Build

**Pull request merged**